bundles/binding/org.openhab.binding.smarthomatic/tst/org/openhab/binding/smarthomatic/TestSHCMessage.java

```diff
@@ -0,0 +1,364 @@
+/**
+ * Copyright (c) 2010-2015, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.smarthomatic;
+
+import java.io.File;
+import java.util.List;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.Unmarshaller;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.openhab.binding.smarthomatic.internal.SHCMessage;
+import org.openhab.binding.smarthomatic.internal.packetData.Packet;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.types.Type;
+
+import junit.framework.Assert;
+
+/**
+ * test of the different smarthomatic messages to make sure they are parsed
+ * correctly
+ * 
+ * @author jbolay
+ *
+ */
+public class TestSHCMessage {
+
+	private Packet packet;
+
+	/**
+	 * common setup
+	 * 
+	 * @throws Exception
+	 */
+	@Before
+	public void setUp() throws Exception {
+		File file = new File(

```

Comment:
Please put your test file in src/test/resources and load it via ResourceAsStream()